packages/messaging-types/index.d.ts

```diff
@@ -25,7 +25,10 @@ import {
 } from '@firebase/util';
 
 export class FirebaseMessaging {
-  private constructor();
+  private constructor(app: FirebaseApp);
+  readonly app: FirebaseApp;
```

Comment:
I don't know Typescript or this codebase that well but I noticed both classes that implement this have a constructor that takes a `FirebaseInternalDependencies` and don't have a top level `app` property, is this type generalized for another use case I'm missing?

Reply:
Thanks for taking a look!

My understanding was that all Firebase services need to have an `app` field (because [this test](https://github.com/firebase/firebase-js-sdk/blob/7250d25145f27ee099dc717a0130c2f8dc9841d2/integration/shared/namespaceDefinition.json#L212) exists), so I added it here, not realizing that it is already defined in the `FirebaseService` interface [here](https://github.com/firebase/firebase-js-sdk/blob/7250d25145f27ee099dc717a0130c2f8dc9841d2/packages/app-types/private.d.ts#L38).

I changed the two classes to implement that interface instead, and removed the `app` field from this one. Please let me know if `app` (and also `INTERNAL`) isn't actually required in the core packages (there's no type checking for it AFAICT), in which case I'll remove it, since it's not actually part of the FCM API.

And I have no idea what happened to that constructor, that's definitely a mistake. :sweat_smile:

Reply:
As far as I can tell INTERNAL and app aren't required in the code anywhere, there is one place that `service.INTERNAL.delete()` is called on services, but it checks to see that `INTERNAL` exists first. I'm not sure if we should remove the `app` getter since it seems like a staple that maybe is useful for debugging or some future plan I don't know about, and it doesn't cost much.

Reply:
Alright, thanks for checking, I'll keep it then. :slightly_smiling_face: